Azure/azure-service-operator
# Support Operator to Add/delete User for PostgreSQL database

Closes #1056 

**What this PR does / why we need it**:
Support the function to add users with specific roles to Postgresql database. 
--Be able to add a user with a predefined roles and delete the user. 
-- Able to delete a user under the postgresql database. 

**Special notes for your reviewer**:
The Azure Database for PostgreSQL server is created with the 3 default roles defined. 

--azure_pg_admin
--azure_superuser
--your server admin user

The server admin user account can be used to create additional users and grant those users into the azure_pg_admin role.
Also, the server admin account can be used to create less privileged users and roles that have access to individual databases and schemas but this function is not included in this PR.  

To verify the PR:
--Use the config/samples/azure_v1alpha1_postgresql_everything.yaml to config pgresql server, database and firewallrule. Ensure at least your local client has the access to azure. 
-- Apply the sample file config/samples/azure_v1alpha1_postgresqluser.yaml to add a user with a predefined role. 
Use any tool to connect the postgresqlServer (e.g. pgadmin). The default server admin password could be derived from kubectl get secret <secret name> -o yaml. By default the secret name is the pgresqlserver name. 
![image](https://user-images.githubusercontent.com/37413937/82871534-516f0e00-9f64-11ea-877d-1c5bf9eec1f6.png)

-- Delete the newly added user by `kubect delete postgresqluser <metatdata name> `. Note the username is the metadata name applied when adding the pgresql user but not the actual pgresql user name. 
From the pgadmin tool, the user has been deleted. 

![image](https://user-images.githubusercontent.com/37413937/82871713-97c46d00-9f64-11ea-9810-aef01a58d9b3.png)



**How does this PR make you feel**:
![gif](https://giphy.com/)

**If applicable**:
